src/metadata.rs

```diff
@@ -224,7 +226,8 @@ impl<'config> MetadataFetcher for CargoInternalsMetadataFetcher<'config> {
       for dependency in package.dependencies().iter() {
         dependencies.push(Dependency {
           name: dependency.name().to_string(),
-          source: dependency.source_id().to_string(),
+          // TODO(bspeice): Justify this unwrap
+          source: serde_json::to_string(dependency.source_id()).unwrap(),
```

Comment:
I assume you're relatively confident about this (if so, LGTM)

Reply:
While I am very confident about that, it wouldn't be terrible to serialize in a `try!()` block before that point. I went ahead and `.unwrap()`'ed it though because the error message would be something along the lines of "Cargo gave us a SourceId that it can't serialize; this is Cargo's fault."